modules/processing/static.py

```diff
@@ -954,16 +956,28 @@ def _parse(self, filepath):
         if len(annoturiset):
             pdfresult["Annot_URLs"] = list(annoturiset)
 
-        return result
+        queue.put(result)
 
     def run(self):
         """Run analysis.
         @return: analysis results dict or None.
         """
         if not os.path.exists(self.file_path):
             return None
+        # Run the analysis in a separate process and impose a timeout
+        q = multiprocessing.Queue()
+        p = multiprocessing.Process(target=self._parse, args=(q, self.file_path))
         log.debug("Starting to load PDF")
-        results = self._parse(self.file_path)
+        p.start()
+        p.join(timeout=self.pdf_timeout)
+        if p.is_alive():
+            log.debug("PDF timeout exceeded, terminating analysis")
+            p.terminate()
+        try:
+            results = q.get(block=False)
+        except:
+            log.debug("No results from PDF analysis")
+            results = {"pdf":{}}
```

Comment:
Should probably put some sort of warning/error here to be displayed in Django. Would help with collecting samples that are giving peepdf an issue.